docs/README.md

```diff
@@ -0,0 +1,423 @@
+
+call_api() {
+   curl -s https://api.example.com/v1/method --url-query "q=$@" \
+      -H "Authorization: Bearer ${EXAMPLE_API_KEY}"
+}
+
+# ...
+```
+
+## Operations and Maintenance
+
+This section discusses some common service operation and maintenance tasks that
+interact with secrets management, and how they interact with setec.
+
+### Secret Rotation
+
+Sometimes secrets need to be rotated. For example, if an API credential expires
+or is compromised, the programs that depend on that credential will need to be
+updated to use a new value.
+
+The simplest way to rotate secrets managed by setec is for the operator to
+install a new version of the secret, mark that version as active, and restart
+or redeploy the programs that use that secret so they will pick up the latest
+version. For example, using the `setec` command-line tool, suppose we have:
+
+```bash
+# HTTP: POST /api/info
+setec info dev/hello-world
+```
+
+This will print the name of the secret, along with the version numbers that
+exist and which one is active:
+
+```
+Name:           dev/hello-world
+Active version: 7
+Versions:       1, 2, 3, 4, 5, 6, 7
+```
+
+In this example we see that the `dev/hello-world` secret has 7 versions and
+version 7 is currently active. To add a new version, use `setec put`:
+
+```bash
+# HTTP: POST /api/put
+setec put dev/hello-world
+```
+
+This will prompt you for the new secret version:
+
+```
+Enter secret: ****
+Confirm secret: ****
+Secret saved as "dev/hello-world", version 8
+```
+
+At this point, version 8 exists, but version 7 is still active. To activate
+version 8, write:
+
+```bash
+# HTTP: POST /api/activate
+setec activate dev/hello-world 8
+```
+
+Now, any client that fetches the active version of `dev/hello-world` will get
+this new value instead.
+
+### Automatic Updates
+
+The example above shows how to simply rotate a secret, but that still requires
+all the programs which depend on that secret to be restarted or redeployed to
+pick up the new value. That is fine for batch processing or low-volume services
+that do not receive a lot of traffic, but may be disruptive for servers with a
+larger active query load.
+
+A program that does not wish to restart to pick up new secret values can poll
+the secrets API periodically, to see whether a new version is available. To do
+this, the request to the `/api/get` method may include the current version the
+client is using, and set the `"updateIfChanged"` flag to `true`:
+
+```json
+{"name":"dev/hello-world", "version":7, "updateIfChanged":true}
+```
+
+If the current active version of the secret still matches what the client
+requested, the server will report 304 Not Modified, indicating to the client
+that it still (already) has the active version. This allows the client to check
+for an update without sending secret values over the wire except when needed,
+and does not trigger an auditable access record unless the value or the
```

Comment:
Agreed, that is better. Done. I left in the bit about permission change, because we also audit log if the caller doesn't have permission for the secret (even if they did previously).